x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/indexer/metrics/MetricField.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.rollup.v2.indexer.metrics;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.xpack.core.rollup.job.MetricConfig;
+import org.elasticsearch.xpack.rollup.v2.FieldValueFetcher;
+import org.elasticsearch.xpack.rollup.v2.indexer.metrics.MetricCollector.Max;
+import org.elasticsearch.xpack.rollup.v2.indexer.metrics.MetricCollector.Min;
+import org.elasticsearch.xpack.rollup.v2.indexer.metrics.MetricCollector.Sum;
+import org.elasticsearch.xpack.rollup.v2.indexer.metrics.MetricCollector.ValueCount;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class MetricField {
```

Comment:
In this PR, I don't add the AggregateDoubleMetricField child class, as I will modify some code of AggregateDoubleMetric, I will add the class, and support `aggregate_metric_double` rollup in the next PR.